Implemented @dehli's occluded vision algorithm

thoughtmanifest opened the pull request:
in Clojure

* ref: https://gist.github.com/dehli/b7658fa9e53459376db612ae650310a2
* Also cleaned up pretty-print-arena => pprint-arena
  - Previous version flipped x & y of arena 
* Added executable code in a comment to allow test
  user to see how occluded-arena works.
* occluded-arena assumes that view-dist is equal
  to the count of the x-dimension since it should
  only be passed the limited view arena

thoughtmanifest commented:
@dehli Just a heads up, it seems that there's a slight bug in my code.  If the user is right next to a wall, the algorithm replaces the wall with an unknown square, whereas it should be keeping the wall.  This doesn't happen if the wall is more than one square away from the user.

dehli commented:
Nice work! @thoughtmanifest I'll take a look later tonight and see if I can figure out what's causing the bug. Also something to think about is computing the slopes just once (since the distance the robot can see won't be changing) and use that in future iterations.

thoughtmanifest commented on src/clj/battlebots/utils/arena.clj:
```diff
@@ -91,6 +92,96 @@
+               (update-in [:blocked-pos]
+                          #(if ang-blocked?
+                             (conj % pos) %)))))
+              bmap (angles view-dist)))
```
pull (angles view-dist) call to the outside of the reduces, only needs to be calc'ed once.

thoughtmanifest commented:
Good catch, I added a line note.  That's a vestige from when I converted from nested loops to reduces.  I had planned to refactor that out and forgot before pushing.

oconn commented:
@thoughtmanifest @dehli This is great! So I just started debugging the issue you're seeing. It appears this only happens when a wall is next to a bot at 45°. I'm going to start taking a closer look and see if I can track this down.

@thoughtmanifest I think I learned more Clojure in the past 10 mins from reading your PR than I have in the past week.
